skyportal/handlers/api/source.py

```diff
@@ -752,8 +758,23 @@ def get(self, obj_id):
             queries_issued=0,
         )
 
+        self.push_notification(
+            'Finding chart generation in progress. Download will start soon.'
+        )
+        rez = await IOLoop.current().run_in_executor(None, finder)
+
         filename = rez["name"]
-        image = rez["data"]
+        image = io.BytesIO(rez["data"])
+
+        # Adapted from
+        # https://bhch.github.io/posts/2017/12/serving-large-files-with-tornado-safely-without-blocking/
+        mb = 1024 * 1024 * 1
+        chunk_size = 1 * mb
+        max_file_size = 5 * mb
+        if not (image.getbuffer().nbytes < max_file_size):
+            return self.error(
+                f"Refusing to send files larger than {max_file_size / mb} MB"
```

Comment:
Maybe clip this to 1 sig fig after the decimal place

Reply:
Addressed.